iroh-base/src/hash.rs

```diff
@@ -191,14 +205,89 @@ impl BlobFormat {
 }
 
 /// A hash and format pair
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, MaxSize)]
 pub struct HashAndFormat {
     /// The hash
     pub hash: Hash,
     /// The format
     pub format: BlobFormat,
 }
 
+#[cfg(feature = "redb")]
+mod redb_support {
+    use super::{Hash, HashAndFormat};
+    use postcard::experimental::max_size::MaxSize;
+    use redb::{RedbKey, RedbValue};
+
+    impl RedbValue for Hash {
+        type SelfType<'a> = Self;
+
+        type AsBytes<'a> = &'a [u8; 32];
+
+        fn fixed_width() -> Option<usize> {
+            Some(32)
+        }
+
+        fn from_bytes<'a>(data: &'a [u8]) -> Self::SelfType<'a>
+        where
+            Self: 'a,
+        {
+            let contents: &'a [u8; 32] = data.try_into().unwrap();
+            (*contents).into()
+        }
+
+        fn as_bytes<'a, 'b: 'a>(value: &'a Self::SelfType<'b>) -> Self::AsBytes<'a>
+        where
+            Self: 'a,
+            Self: 'b,
+        {
+            value.as_bytes()
+        }
+
+        fn type_name() -> redb::TypeName {
+            redb::TypeName::new("iroh_base::Hash")
+        }
+    }
+
+    impl RedbKey for Hash {
+        fn compare(data1: &[u8], data2: &[u8]) -> std::cmp::Ordering {
+            data1.cmp(&data2)
+        }
+    }
+
+    impl RedbValue for HashAndFormat {
+        type SelfType<'a> = Self;
+
+        type AsBytes<'a> = [u8; Self::POSTCARD_MAX_SIZE];
+
+        fn fixed_width() -> Option<usize> {
+            Some(33)
+        }
+
+        fn from_bytes<'a>(data: &'a [u8]) -> Self::SelfType<'a>
+        where
+            Self: 'a,
+        {
+            let t: &'a [u8; Self::POSTCARD_MAX_SIZE] = data.try_into().unwrap();
+            postcard::from_bytes(t.as_slice()).unwrap()
```

Comment:
This unwrap is a bit unnerving, but the RedbValue types does not give us any other option.